Tests/DatadogIntegrationTests/RUMIntegrationTests.swift

```diff
@@ -48,30 +48,59 @@ class RUMIntegrationTests: IntegrationTests {
         let rumEventsMatchers = try recordedRUMRequests
             .flatMap { request in try RUMEventMatcher.fromNewlineSeparatedJSONObjectsData(request.httpBody) }
 
-        // Assert `application_start` action
+        print(rumEventsMatchers.count)
+
+        // Assert Fixture 1 VC ⬇️
+
+        // ----> `application_start` Action due to initial startView()
         let applicationStartAction: RUMActionEvent = try rumEventsMatchers[0].model()
         XCTAssertEqual(applicationStartAction.action.type, "application_start")
 
-        // Assert Fixture 1 View updates
+        // ----> View update on startView()
         let view1UpdateA: RUMViewEvent = try rumEventsMatchers[1].model()
         XCTAssertEqual(view1UpdateA.dd.documentVersion, 1)
         XCTAssertEqual(view1UpdateA.view.action.count, 1)
+        XCTAssertEqual(view1UpdateA.view.resource.count, 0)
 
+        // --------> View update on stopResourceLoading()
         let view1UpdateB: RUMViewEvent = try rumEventsMatchers[2].model()
+        XCTAssertEqual(view1UpdateB.view.id, view1UpdateA.view.id)
         XCTAssertEqual(view1UpdateB.dd.documentVersion, 2)
         XCTAssertEqual(view1UpdateB.view.action.count, 1)
+        XCTAssertEqual(view1UpdateB.view.resource.count, 1)
+
+        // --------> Resource event on stopResourceLoading()
+        let resourceLoaded: RUMResourceEvent = try rumEventsMatchers[3].model()
+        XCTAssertEqual(resourceLoaded.view.id, view1UpdateA.view.id)
+        XCTAssertEqual(resourceLoaded.resource.url, "https://foo.com/resource/1")
+        XCTAssertEqual(resourceLoaded.resource.statusCode, 200)
+        XCTAssertEqual(resourceLoaded.resource.type, "other")
+        XCTAssertGreaterThan(resourceLoaded.resource.duration, 100_000_000 - 1) // ~0.1s
+        XCTAssertLessThan(resourceLoaded.resource.duration, 100_000_000 * 3) // less than 0.3s
 
-        // Assert Fixture 2 View updates
-        let view2UpdateA: RUMViewEvent = try rumEventsMatchers[3].model()
+        // ----> View update on stopView()
+        let view1UpdateC: RUMViewEvent = try rumEventsMatchers[4].model()
+        XCTAssertEqual(view1UpdateC.view.id, view1UpdateA.view.id)
+        XCTAssertEqual(view1UpdateC.dd.documentVersion, 3)
+        XCTAssertEqual(view1UpdateC.view.action.count, 1)
+        XCTAssertEqual(view1UpdateC.view.resource.count, 1)
```

Comment:
I don't see where you test that the event is delegated to children also (actions, resources). They should be stopped too if the view was stopped.

Reply:
Yes, this is additional condition to handle (`View` is not stopped until all its `Resources` complete). To keep this PR small, I separated this to `RUM-635 Resource loading can exceed View's lifespan` - this will be done later👌.